src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceAuthorizationBehavior.cs

```diff
@@ -146,14 +141,13 @@ public bool ImpersonateOnSerializingReply
             }
         }
 
-        public IAuthorizationService AuthorizationService
+        public IServiceScopeFactory ServiceScopeFactory
         {
```

Comment:
This is a public type, we can't remove public API once we've released it. There's a few options here.  
1. The api could be marked as obsolete  
2. The property could act like a factory where it instantiates a scope and returns a new instance.  

If the second option works, it's my preference. Regardless any new properties should be internal.

Reply:
`AuthorizationService` property now acts as a factory. I marked it obsolete as I should have kept it internal in original PR. Please notice `ServiceAuthorizationBehavior` now implements `IDisposable` as I need to dispose the created `IServiceScope`.